### ISSUE TITLE
OMDB HTTPError: 401 Client Error

### ISSUE THREAD

**Issue by pannal** [Diaoul/subliminal#759]
By-proxy for @ukdtom. Happens for him every time. Will check locally.

```
2017-05-20 20:01:37,359 (7f7706c91700) :  ERROR (logger:38) - subliminal_patch.core: Failed to refine video: Traceback (most recent call last):
  File "/share/CACHEDEV1_DATA/.qpkg/PlexMediaServer/Library/Plex Media Server/Plug-ins/Sub-Zero.bundle/Contents/Libraries/Shared/subliminal_patch/core.py", line 577, in refine
    refiner_manager[refiner].plugin(video, **kwargs)
  File "/share/CACHEDEV1_DATA/.qpkg/PlexMediaServer/Library/Plex Media Server/Plug-ins/Sub-Zero.bundle/Contents/Libraries/Shared/subliminal/refiners/omdb.py", line 155, in refine
    results = search(video.title, 'movie', video.year)
  File "/share/CACHEDEV1_DATA/.qpkg/PlexMediaServer/Library/Plex Media Server/Plug-ins/Sub-Zero.bundle/Contents/Libraries/Shared/dogpile/cache/region.py", line 1220, in decorate
    should_cache_fn)
  File "/share/CACHEDEV1_DATA/.qpkg/PlexMediaServer/Library/Plex Media Server/Plug-ins/Sub-Zero.bundle/Contents/Libraries/Shared/dogpile/cache/region.py", line 825, in get_or_create
    async_creator) as value:
  File "/share/CACHEDEV1_DATA/.qpkg/PlexMediaServer/Library/Plex Media Server/Plug-ins/Sub-Zero.bundle/Contents/Libraries/Shared/dogpile/lock.py", line 154, in __enter__
    return self._enter()
  File "/share/CACHEDEV1_DATA/.qpkg/PlexMediaServer/Library/Plex Media Server/Plug-ins/Sub-Zero.bundle/Contents/Libraries/Shared/dogpile/lock.py", line 94, in _enter
    generated = self._enter_create(createdtime)
  File "/share/CACHEDEV1_DATA/.qpkg/PlexMediaServer/Library/Plex Media Server/Plug-ins/Sub-Zero.bundle/Contents/Libraries/Shared/dogpile/lock.py", line 145, in _enter_create
    created = self.creator()
  File "/share/CACHEDEV1_DATA/.qpkg/PlexMediaServer/Library/Plex Media Server/Plug-ins/Sub-Zero.bundle/Contents/Libraries/Shared/dogpile/cache/region.py", line 792, in gen_value
    created_value = creator()
  File "/share/CACHEDEV1_DATA/.qpkg/PlexMediaServer/Library/Plex Media Server/Plug-ins/Sub-Zero.bundle/Contents/Libraries/Shared/dogpile/cache/region.py", line 1216, in creator
    return fn(*arg, **kw)
  File "/share/CACHEDEV1_DATA/.qpkg/PlexMediaServer/Library/Plex Media Server/Plug-ins/Sub-Zero.bundle/Contents/Libraries/Shared/subliminal/refiners/omdb.py", line 76, in search
    results = omdb_client.search(title, type, year)
  File "/share/CACHEDEV1_DATA/.qpkg/PlexMediaServer/Library/Plex Media Server/Plug-ins/Sub-Zero.bundle/Contents/Libraries/Shared/subliminal/refiners/omdb.py", line 59, in search
    r.raise_for_status()
  File "/share/CACHEDEV1_DATA/.qpkg/PlexMediaServer/Library/Plex Media Server/Plug-ins/Sub-Zero.bundle/Contents/Libraries/Shared/requests/models.py", line 909, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
HTTPError: 401 Client Error: Unauthorized for url: http://www.omdbapi.com/?r=json&v=1&y=2016&s=Allied&type=movie&page=1
```

**Comment by pannal** [comment]
DId they just shut out our User-Agent?

**Comment by pannal** [comment]
Uh, the response is `{"Response":"False","Error":"No API key provided."}`

**Comment by pannal** [comment]
Yeah, you gotta be a patreon of them to get an API key.

**Comment by pannal** [comment]
Hit me up on IRC, mate.

**Comment by fernandog** [comment]
@Diaoul how to make OMDB optional and accept the api key? IMHO need to handle this before merging into master

**Comment by fernandog** [comment]
@shlomiLan look at this

**Comment by shlomiLan** [comment]
Is anyone working on this issue? 
I saw that they have a [free](http://www.omdbapi.com/apikey.aspx) keys, but it has a limit of 1000 daily requests. should this be a problem?

**Comment by fernandog** [comment]
Too low. It should be default to off and user provide an API key

**Comment by Fossil01** [comment]
Any progress on this? You can get a free API key from OMDB.

**Comment by marcelpaulo** [comment]
Shouldn't the `omdb` refiner be turned off by default, as @fernandog suggested ? This issue will keep on biting whoever uses subliminal without explicitly asking for the other refiners on the command line (it just bit me).

**Comment by talonius** [comment]
Quick fix for anyone else who has this issue.  Modify omdb.py, add

```
def get(self, id=None, title=None, type=None, year=None, plot='short', tomatoes=False):
...
     params['tomatoes'] = tomatoes
     params['apikey'] = '<api key>'
...
```

and 'apikey': '<api key>' to the params list at the beginning of the search function.

```
def search(self, title, type=None, year=None, page=1):
     params ={'s': title, 'type': type, 'y': year, 'page': page, 'apikey': '<api key>' } 
```

**This is a hack.**  Yes, it's hard coded.  There seemed to be some confusion above as to how to properly address this so I didn't write a proper patch, or test very thoroughly.  **It works for me.**

**Comment by ratoaq2** [comment]
Fixed by 3aea2d7d71b6a5c055104b526a75fe6b78d1cbb1

Now you can provide the apikey when using subliminal.
If not provided, refiner will be skipped

**Comment by euphoria360** [comment]
> Fixed by [3aea2d7](https://github.com/Diaoul/subliminal/commit/3aea2d7d71b6a5c055104b526a75fe6b78d1cbb1)
> 
> Now you can provide the apikey when using subliminal.
> If not provided, refiner will be skipped

how can I provide my omdb API key? I couldn't find where should I add it using subliminal CLI.

**Comment by basilky** [comment]
@euphoria360  You can provide it using --omdb "APIKEY"
